### PR TITLE
Introduce CacheSectionHelper

### PR DESCRIPTION
**Pull request by mkouba**
- configure and register the cached section automatically if the quarkus-cache extension is present

### CODE REVIEW

**Comment by mkouba** [comment]
@gwenneg Just FYI :-)

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -1242,6 +1242,49 @@ The behavior is very similar to the <<include_helper>> but:
 
 NOTE: The evaluated template is parsed and evaluated every time the section is executed. In other words, it's not possible to cache the parsed value to conserve resources and optimize the performance.
 
+==== Cached Section
+
+Sometimes it's practical to cache parts of the template that rarely change.
+In order to use the caching capability you'll need to register and configure the built-in `io.quarkus.qute.CacheSectionHelper.Factory`:
```
Comment (gastaldi) [inline docs/src/main/asciidoc/qute-reference.adoc:1248]: Can `@EngineConfiguration` be used here?

**Review by gastaldi — APPROVED** [review]

File: docs/src/main/asciidoc/qute-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/qute-reference.adoc:1248]: Well, you would have to extend the original factory class. Anyway, it should not be needed if the  quarkus-cache extension is used...

**Comment by mkouba** [comment]
It seems that the failures in the `integration-tests/devtools` module are not related.

**Pull request merged**